### ISSUE TITLE
Issue with IO and `<internal:prelude>`

### ISSUE THREAD

**Issue by deecewan** [Shopify/tapioca#44]
I've recently run into an issue where the symbol generator is trying to do `File.realpath` on `<internal:prelude>` and therefore crashing.

Can't exactly pinpoint what broke this behaviour, but it's now broken :( It may be because a gem is now updating/modifying parts of `IO` 🤷‍♂ 

Also just seen it with `(eval)` as the path, too. Can we maybe do a `File.exists?` or similar before `File.realpath` to make sure it is there instead of blowing up?

Below is the stack trace

```
Compiling hamster, this may take a few seconds... bundler: failed to load command: tapioca (/Users/david/.rbenv/versions/2.6.5/bin/tapioca)
Errno::ENOENT: No such file or directory @ realpath_rec - /Users/david/projects/ferocia/up/api/<internal:prelude>
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:429:in `realpath'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:429:in `path_in_gem?'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:465:in `method_in_gem?'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:389:in `compile_method'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:355:in `block (2 levels) in compile_directly_owned_methods'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:353:in `map'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:353:in `block in compile_directly_owned_methods'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:352:in `each'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:352:in `flat_map'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:352:in `compile_directly_owned_methods'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:336:in `compile_methods'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:177:in `block in compile_body'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:450:in `with_indentation'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:176:in `compile_body'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:162:in `compile_module'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:114:in `compile_constant'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:100:in `compile'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:74:in `generate_from_symbol'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:221:in `call'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:221:in `block in _on_method_added'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:32:in `map'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table/symbol_generator.rb:32:in `generate'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `call'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `block in _on_method_added'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/compilers/symbol_table_compiler.rb:19:in `compile'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `call'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `block in _on_method_added'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/generator.rb:285:in `compile_rbi'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `call'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `block in _on_method_added'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/generator.rb:234:in `block (2 levels) in perform_additions'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/generator.rb:225:in `each'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/generator.rb:225:in `block in perform_additions'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/shell/basic.rb:44:in `indent'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/generator.rb:219:in `perform_additions'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `call'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `block in _on_method_added'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/generator.rb:74:in `sync_rbis_with_gemfile'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `call'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.5478/lib/types/private/methods/_methods.rb:240:in `block in _on_method_added'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/cli.rb:63:in `block in sync'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca.rb:10:in `silence_warnings'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/lib/tapioca/cli.rb:62:in `sync'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
  /Users/david/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/tapioca-0.2.8/exe/tapioca:6:in `<top (required)>'
  /Users/david/.rbenv/versions/2.6.5/bin/tapioca:23:in `load'
  /Users/david/.rbenv/versions/2.6.5/bin/tapioca:23:in `<top (required)>'
```

**Comment by paracycle** [comment]
@deecewan Apologies for the breaking change in version 0.2.8 that ended up breaking some of the legitimate use cases like this. We have released 0.3.0 that includes [the fix](https://github.com/Shopify/tapioca/pull/40) for that problem. Can you test using that version and reopen the issue if it is still failing for you? Thanks.

**Comment by deecewan** [comment]
@paracycle thanks! fixed :)
